Mathlib/Analysis/Distribution/ContDiffMapSupportedIn.lean

```diff
@@ -159,4 +159,204 @@ theorem coe_copy (f : 𝓓^{n}_{K}(E, F)) (f' : E → F) (h : f' = f) : ⇑(f.co
+   SMul R 𝓓^{n}_{K}(E, F) where
+  smul c f := ContDiffMapSupportedIn.mk (c • (f : E → F)) (f.contDiff.const_smul c) <| by
+    rw [← smul_zero c]
+    exact f.zero_on_compl.comp_left
+
+@[simp]
+lemma coe_smul {R} [Semiring R] [Module R F] [SMulCommClass ℝ R F] [ContinuousConstSMul R F]
+    (c : R) (f : 𝓓^{n}_{K}(E, F)) : (c • f : 𝓓^{n}_{K}(E, F)) = c • (f : E → F) :=
+  rfl
+
+@[simp]
+lemma smul_apply {R} [Semiring R] [Module R F] [SMulCommClass ℝ R F] [ContinuousConstSMul R F]
+    (c : R) (f : 𝓓^{n}_{K}(E, F)) (x : E) : (c • f) x = c • (f x) :=
+  rfl
+
+instance : AddCommGroup 𝓓^{n}_{K}(E, F) :=
+  DFunLike.coe_injective.addCommGroup _ rfl (fun _ _ => rfl) (fun _ => rfl) (fun _ _ => rfl)
+    (fun _ _ => rfl) fun _ _ => rfl
+
+variable (E F K n)
+
+/-- Coercion as an additive homomorphism. -/
+def coeHom : 𝓓^{n}_{K}(E, F) →+ E → F where
+  toFun f := f
+  map_zero' := coe_zero
+  map_add' _ _ := rfl
+
+variable {E F}
+
+theorem coe_coeHom : (coeHom E F n K : 𝓓^{n}_{K}(E, F) → E → F) = DFunLike.coe :=
+  rfl
+
+theorem coeHom_injective : Function.Injective (coeHom E F n K) := by
+  rw [coe_coeHom]
+  exact DFunLike.coe_injective
+
+end AddCommGroup
+
+section Module
+
+instance {R} [Semiring R] [Module R F] [SMulCommClass ℝ R F] [ContinuousConstSMul R F] :
+    Module R 𝓓^{n}_{K}(E, F) :=
+  (coeHom_injective n K).module R (coeHom E F n K) fun _ _ => rfl
+
+end Module
+
+protected theorem support_subset (f : 𝓓^{n}_{K}(E, F)) : support f ⊆ K :=
+  support_subset_iff'.mpr f.zero_on_compl
+
+protected theorem tsupport_subset (f : 𝓓^{n}_{K}(E, F)) : tsupport f ⊆ K :=
+  closure_minimal f.support_subset K.isCompact.isClosed
+
+protected theorem hasCompactSupport (f : 𝓓^{n}_{K}(E, F)) : HasCompactSupport f :=
+  HasCompactSupport.intro K.isCompact f.zero_on_compl
+
+/-- Inclusion of unbundled `n`-times continuously differentiable function with support included
+in a compact `K` into the space `𝓓^{n}_{K}`. -/
+protected def of_support_subset {f : E → F} (hf : ContDiff ℝ n f) (hsupp : support f ⊆ K) :
+    𝓓^{n}_{K}(E, F) where
+  toFun := f
+  contDiff' := hf
+  zero_on_compl' := support_subset_iff'.mp hsupp
+
+protected theorem bounded_iteratedFDeriv (f : 𝓓^{n}_{K}(E, F)) {i : ℕ} (hi : i ≤ n) :
+    ∃ C, ∀ x, ‖iteratedFDeriv ℝ i f x‖ ≤ C :=
+  Continuous.bounded_above_of_compact_support
+    (f.contDiff.continuous_iteratedFDeriv <| (WithTop.le_coe rfl).mpr hi)
+    (f.hasCompactSupport.iteratedFDeriv i)
+
+
+/-- Inclusion of `𝓓^{n}_{K}(E, F)` into the space `E →ᵇ F` of bounded continuous maps
+as a `𝕜`-linear map. -/
+@[simps]
+noncomputable def toBoundedContinuousFunctionₗ : 𝓓^{n}_{K}(E, F) →ₗ[𝕜] E →ᵇ F  where
+  toFun f := f
+  map_add' _ _ := rfl
+  map_smul' _ _ := rfl
+
+/-- Wrapper for `iteratedFDeriv i` on `𝓓^{n}_{K}(E, F)`,
+as a map into `𝓓^{n-i}_{K}(E, E [×i]→L[ℝ] F)`. -/
+noncomputable def iteratedFDerivWithOrder (i : ℕ) (f : 𝓓^{n}_{K}(E, F)) :
```

Comment:
There's a more interesting design question: should you use the current convention (you can ask for as many derivatives as you want, but asking for too many always gives zero) --- or would it be better to define this make from D^{n+i} to D^n instead? The current definition includes a branch and natural number subtraction; I suspect that won't be as nice to work with.

What are your thoughts on this so far? (In the mean-time, let me compare with other parts of the calculus library...)

Reply:
Comparing with ContDiff.iteratedFDeriv_right and ContDiff.iteratedFDeriv_right', I tend towards making this a definition on D^{n+i} to D^n.